c_emulator/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+set(EMULATOR_COMMON_SRCS
+    riscv_config.h
+    riscv_platform.c
+    riscv_platform.h
+    riscv_platform_impl.c
+    riscv_platform_impl.h
+    riscv_prelude.c
+    riscv_prelude.h
+    riscv_sail.h
+    riscv_sim.c
+    riscv_softfloat.c
+    riscv_softfloat.h
+)
+
+foreach (xlen IN ITEMS 32 64)
+    foreach (flen IN ITEMS 32 64)
+        foreach (variant IN ITEMS "" "rvfi")
+            set(arch "rv${xlen}")
+            if (flen EQUAL 32)
+                string(APPEND arch "f")
+            else()
+                string(APPEND arch "d")
+            endif()
+            if (variant)
+                string(APPEND arch "_${variant}")
+            endif()
+
+            add_executable(c_emulator_${arch}
+                "${CMAKE_BINARY_DIR}/riscv_model_${arch}.c"
+                ${EMULATOR_COMMON_SRCS}
+            )
+
+            if (NOT arch IN_LIST DEFAULT_ARCHITECTURES)
+                set_target_properties(c_emulator_${arch} PROPERTIES EXCLUDE_FROM_ALL TRUE)
+            endif()
+
+            add_dependencies(c_emulator_${arch} generated_model_${arch})
+
+            target_link_libraries(c_emulator_${arch}
+                PRIVATE softfloat sail_runtime GMP::GMP ZLIB::ZLIB
+            )
+
+            target_include_directories(c_emulator_${arch}
+                # So the generated C can find riscv_platform/prelude.h"
+                PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}"
+            )
+
+            if (arch MATCHES "rvfi")
+                target_compile_definitions(c_emulator_${arch}
+                    PRIVATE RVFI_DII
+                )
+            endif()
+
+            # TODO: Enable warnings when we use the #include trick
+            # to include the generated Sail code. Currently it
+            # generates too many warnings to turn these on globally.
+
+            # target_compile_options(c_emulator_${arch} PRIVATE
+            #     -Wall -Wextra
```

Comment:
IMO we should enable -Wall globally in the root CMakeLists.txt. Possibly even -Wextra although I imagine the softfloat library will not be happy, so that probably makes more sense to have just here.

Reply:
Yeah it's not just softfloat though (which isn't even happy with `-Wall`) - the `model.c` also generates a load of warnings. Once we've removed softfloat, and use the `#include "model.c"` trick for the model we can though.